tests/src/gui/testqgsrelationreferencewidget.cpp

```diff
@@ -340,5 +341,38 @@ void TestQgsRelationReferenceWidget::testSetGetForeignKey()
   QCOMPARE( spy.count(), 3 );
 }
 
+
+// Test issue https://issues.qgis.org/issues/22071
+// Relation reference widget wrong feature when "on map identification"
+void TestQgsRelationReferenceWidget::testIdentifyOnMap()
+{
+  QWidget parentWidget;
+  QgsRelationReferenceWidget w( &parentWidget );
+  QVERIFY( mLayer1->startEditing() );
+  w.setRelation( *mRelation, true );
+  w.setAllowMapIdentification( true );
+  w.init();
+  QEventLoop loop;
+  // Populate model (I tried to listen to signals but the module reload() runs twice
+  // (the first load triggers a second one which does the population of the combo)
+  // and I haven't fin a way to properly wait for it.
+  QTimer::singleShot( 300, [&] { loop.quit(); } );
+  loop.exec();
+  QgsFeature feature;
+  mLayer2->getFeatures( QStringLiteral( R"(pk = %1)" ).arg( 11 ) ).nextFeature( feature );
```

Comment:
Is there a particular reason for the `R`aw string here?

Reply:
no, I kind of use it all the times to avoid quoting and escaping, but in this case a normal string would equally do.

Reply:
Ok, I just thought I missed some sort of new "best practice", I haven't seen much of them in the code before.